#6 New Feature added

eMKey1988 opened the pull request:
Added saving of State and ability to save Custom Maps

eMKey1988 committed:
New Feature added

Added saving of State and ability to save Custom Maps

uberswe commented:
Thank you so much for the PR! I will have a look at this tomorrow.

Fixes #4 and #6.

uberswe commented on TarkovMapOverlay/MainWindow.xaml:
```diff
@@ -6,12 +6,10 @@
         BorderBrush="Black"
         BorderThickness="1"
-        Background="Black"
-        AllowsTransparency="True">
```
I think that removing `AllowsTransparency` has broken the ability to make the background transparent or to change to opacity. The rest looks good to me and I tested the saving and loading which works well. So just fix this part and I will merge and make a release. Thanks!

![be5fcc2613a3088158cdf2896cb38766](https://user-images.githubusercontent.com/1903284/107186466-a06ead80-69e4-11eb-962b-d9ad48372867.png)

eMKey1988 replied:
oh sorry for that, that was a mistake. i didn't notices that i changed something there.

eMKey1988 committed:
Mousefeature Added + fixes

-Mouse-minimizing feature Added
-Key and Mousebind get now saved
-Fixed issue with backgroundopacity

eMKey1988 committed:
Adding another savingvariables

Adding saving of the Window size and Position

eMKey1988 committed:
new maps + code cleaning

just added a few resources for later use if needed

uberswe commented on TarkovMapOverlay/MainWindow.xaml.cs:
```diff
@@ -201,5 +324,130 @@ private void MinimizeKeybind_OnClick(object sender, RoutedEventArgs e)
+
+            TarkovMap.Source = transformBmp;
+        }
+        */
```
please remove any commented code

uberswe commented on TarkovMapOverlay/MainWindow.xaml.cs:
```diff
@@ -22,23 +24,119 @@ public partial class MainWindow : Window
+            {
+                try
+                {
+                    BitmapImage bitmap = new BitmapImage(new Uri(settings.currentMapPath, UriKind.RelativeOrAbsolute));
```
This line is giving me an error.

```
Argument 2: cannot convert from 'System.UriKind' to 'bool'
```

uberswe replied:
```suggestion
                    BitmapImage bitmap = new BitmapImage(new Uri(settings.currentMapPath, uriKind: UriKind.RelativeOrAbsolute));
```
I think adding `uriKind: ` as an argument fixes this issue.

uberswe commented on TarkovMapOverlay/MainWindow.xaml.cs:
```diff
@@ -201,5 +324,130 @@ private void MinimizeKeybind_OnClick(object sender, RoutedEventArgs e)
+            return new SavedSettings();
+        }
+
+        public void MoveIntoView() //important if saved WindowPosition is out of screenbounds ( if saved on 2.screen for example )
+        {
+            double _windowTop = this.Top;
+            double _windowHeight = this.Height;
+            double _windowLeft = this.Left;
+            double _windowWidth = this.Width;
+            
+            if (_windowTop + _windowHeight / 2 > SystemParameters.VirtualScreenHeight)
+            {
+                _windowTop = SystemParameters.VirtualScreenHeight -_windowHeight;
+            }
+
+            if (_windowLeft + _windowWidth / 2 > SystemParameters.VirtualScreenWidth)
+            {
+                _windowLeft = SystemParameters.VirtualScreenWidth - _windowWidth;
+            }
+
+            if (_windowTop < 0)
+            {
+                _windowTop = 0;
+            }
+
+            if (_windowLeft < 0)
+            {
+                _windowLeft = 0;
+            }
+
+            this.Top = _windowTop;
+            this.Height = _windowHeight;
+            this.Left = _windowLeft;
+            this.Width = _windowWidth;
+        }
```
For some reason the window opens with a 0 by 0 size if there are no settings saved previously and it's very hard to find 😄 

Also, it may be good if the window doesn't start in the top left of the screen if there are no settings set.

eMKey1988 replied:
i think this relates to the fact that the SavedSettings class is missing somehow. XD ( 0x0 size )
do you have a screenshot from the Windows startposition? Normally it should be at 200,200 if it's saved out of screen when you had a second monitor and now just one.

uberswe replied:
it was in the very top left corner. I think it may have been because I had a save file already from previous testing.

eMKey1988 replied:
I cannot reproduce this behaviour. maybe just because the SavedSettings - class was missed the first time, and it defaults to 0 in this case.

uberswe reviewed (changes requested):
Great work, the settings seem to load and save well. The transparency issue is fixed. However there are some more issues I noticed now. Please see my comments.

Can you remove the spaces in the following file names? Spaces can sometimes cause issues with file paths so I would prefer to avoid having spaces. An `_` underscore is a better alternative.

 - TarkovMapOverlay/Resources/Customs Dorms.jpeg
 - TarkovMapOverlay/Resources/Customs Stashes.jpeg
 - TarkovMapOverlay/Resources/Woods Stashes.jpeg
 
 You should also add menu items for these maps as I have done here https://github.com/uberswe/Tarkov-Map-Overlay/blob/master/TarkovMapOverlay/MainWindow.xaml.cs#L139-L179
 
I also noticed that your new file `SavedSettings.cs` is not included in the project. Maybe you forgot to add the changes you have made to `TarkovMapOverlay.sln`?

eMKey1988 committed:
fixed some issues

-edited names
-added Eventhandlers to new maplist-items
